actions/config.go

```diff
@@ -68,7 +68,9 @@ var (
 func CreateConfigFile(target time.Duration, useLegacy bool) error {
 	// Create the config file before computing the hashing costs, so we fail
 	// immediately if the program has insufficient permissions.
+	oldMask := unix.Umask(0)
 	configFile, err := os.OpenFile(ConfigFileLocation, createFlags, configPermissions)
+	unix.Umask(oldMask)
```

Comment:
I initially though this should be a `defer` in case `os.OpenFile` panics, but now I realize it's OK, as we shouldn't open any additional files on panic.

Reply:
I decided to change it to just do the OpenFile() in a helper function that overrides umask.